### PR TITLE
Fix all prefer-const reports in JavaScript docs

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

A one-time effort to fix all prefer-const in JavaScript docs!

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/date/tolocalestring/index.md [hunk 1]
```diff
@@ -127,10 +127,15 @@ The results provided by `toLocaleString()` can be customized using the
...
-let date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));
+const date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));
 
 // Request a weekday along with a long date
-let options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+const options = {
+  weekday: 'long',
+  year: 'numeric',
+  month: 'long',
+ day: 'numeric',
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/date/tolocalestring/index.md:137]:
```suggestion
  day: 'numeric',
```

File: files/en-us/web/javascript/reference/global_objects/date/gethours/index.md [hunk 2]
```diff
@@ -36,8 +36,8 @@ based on the value of the {{jsxref("Global_Objects/Date", "Date")}} object
 `Xmas95`.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/date/gethours/index.md:36]: Let's not start variable with capital letters:
```suggestion
`xmas95`.
```

File: files/en-us/web/javascript/reference/global_objects/date/gethours/index.md [hunk 3]
```diff
@@ -36,8 +36,8 @@ based on the value of the {{jsxref("Global_Objects/Date", "Date")}} object
...
 
 ```js
-let Xmas95 = new Date('December 25, 1995 23:15:30');
-let hours = Xmas95.getHours();
+const Xmas95 = new Date('December 25, 1995 23:15:30');
+const hours = Xmas95.getHours();
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/date/gethours/index.md:40]:
```suggestion
const xmas95 = new Date('December 25, 1995 23:15:30');
const hours = xmas95.getHours();
```

File: files/en-us/web/javascript/reference/global_objects/symbol/symbol/index.md [hunk 4]
```diff
@@ -73,8 +73,8 @@ If you really want to create a `Symbol` wrapper object, you can use the
 `Object()` function:
 
 ```js
-let sym    = Symbol('foo');
-let symObj = Object(sym);
+const sym = Symbol('foo');
+const symObj = Object(sym);
 typeof sym    // => "symbol"
 typeof symObj // => "object"
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/symbol/symbol/index.md:79]: No `;` or `console.log()`?

File: files/en-us/web/javascript/reference/global_objects/weakset/index.md [hunk 5]
```diff
@@ -32,27 +32,26 @@ Functions that call themselves recursively need a way of guarding against circul
...
 ```js
 // Execute a callback on everything stored inside an object
-function execRecursively(fn, subject, _refs = null){
-  if(!_refs)
-    _refs = new WeakSet();
-
+function execRecursively(fn, subject, _refs = new WeakSet()) {
   // Avoid infinite recursion
-  if(_refs.has(subject))
+  if(_refs.has(subject)) {
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/weakset/index.md:37]:
```suggestion
  if (_refs.has(subject)) {
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/reference/global_objects/symbol/symbol/index.md (hunk 4 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/symbol/symbol/index.md:79]: We have a lot of places where we are using a bare expression in place of a `console.log()`, where we usually don't add the semi. We can fix that later.

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**